tests/sentry/deletions/test_validate_group_related_models.py

```diff
@@ -0,0 +1,132 @@
+"""
+Tests to ensure all models with group foreign keys are properly registered
+in group deletion configuration.
+"""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+
+import django.apps
+from django.db import models as django_models
+
+from sentry.db.models import FlexibleForeignKey
+from sentry.deletions.defaults.group import (
+    ADDITIONAL_GROUP_RELATED_MODELS,
+    DIRECT_GROUP_RELATED_MODELS,
+)
+from sentry.testutils.cases import TestCase
+
+
+class GroupRelatedModelsCompletenessTest(TestCase):
+    """
+    Validates that all models with group foreign keys are accounted for in the
+    deletion configuration to prevent orphaned records or cascade delete timeouts.
+    """
+
+    # Models that don't need to be in either list (with justification)
+    EXEMPTED_MODELS: Mapping[str, str] = {
+        # Add models here if they shouldn't be in either list, with a comment explaining why
+        # Example: "sentry.SomeModel": "Uses custom deletion logic in XYZ",
+        "sentry.Activity": "TBD",
+        "sentry.GroupReaction": "TBD",
+        "sentry.PlatformExternalIssue": "TBD",
+    }
+
+    def get_models_with_group_foreign_key(self) -> set[type[django_models.Model]]:
+        """
+        Discover all models in the sentry app that have a foreign key to Group.
+        """
+        from sentry.models.group import Group
+
+        models_with_group_fk = set()
+
+        for model_class in django.apps.apps.get_models():
+            # Only check sentry app models
+            if model_class._meta.app_label not in ("sentry", "notifications"):
```

Comment:
Couldn't we end up with gaps if issues models are moved to other django apps? I would expect that the filtering you're doing on `group_id` columns pointing to `Group` should be enough to eliminate any noise.

Reply:
That makes sense. On it.